src/jaxsim/parsers/rod/utils.py

```diff
@@ -170,23 +167,31 @@ def create_sphere_collision(
 
     # From https://stackoverflow.com/a/26127012
     def fibonacci_sphere(samples: int) -> npt.NDArray:
-        points = []
-        phi = np.pi * (3.0 - np.sqrt(5.0))  # golden angle in radians
-
-        for i in range(samples):
-            y = 1 - (i / float(samples - 1)) * 2  # y goes from 1 to -1
-            radius = np.sqrt(1 - y * y)  # radius at y
-
-            theta = phi * i  # golden angle increment
-
-            x = np.cos(theta) * radius
-            z = np.sin(theta) * radius
+        # Get the golden ratio in radians.
+        phi = np.pi * (3.0 - np.sqrt(5.0))
+
+        # Generate the points.
+        points = [
+            np.array(
+                [
+                    np.cos(phi * i)
+                    * np.sqrt(1 - (y := 1 - 2 * i / (samples - 1)) ** 2),
+                    y,
+                    np.sin(phi * i) * np.sqrt(1 - y**2),
+                ]
+            )
+            for i in range(samples)
+        ]
 
-            points.append(np.array([x, y, z]))
+        # Filter to keep only the bottom half if required.
+        if os.environ.get("JAXSIM_COLLISION_USE_BOTTOM_ONLY", "0"):
+            # Keep only the points with y <= 0.
```

Comment:
You are selecting the points based on `point[2]`, there's a typo in the comment then?

```suggestion
            # Keep only the points with z <= 0.
```

Reply:
Yes, thank you @xela-95! Addressed in https://github.com/ami-iit/jaxsim/pull/312/commits/8ab51f35e220754e99b31fb5397dc8e12e8fc4b1